rbac/rbac.go

```diff
@@ -0,0 +1,121 @@
+package rbac
+
+import (
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Permission is an Observatorium RBAC permission.
+type Permission string
+
+const (
+	// Write gives access to write data to a tenant.
+	Write Permission = "write"
+	// Read gives access to read data from a tenant.
+	Read Permission = "read"
+)
+
+// Role describes a set of permissions to interact with a tenant.
+type Role struct {
+	Name        string       `json:"name" yaml:"name"`
+	Tenants     []string     `json:"tenants" yaml:"tenants"`
+	Permissions []Permission `json:"permissions" yaml:"permissions"`
+}
+
+// RoleBinding binds a set of roles to a set of subjects.
+type RoleBinding struct {
+	Name     string   `json:"name" yaml:"name"`
+	Subjects []string `json:"subjects" yaml:"subjects"`
+	Roles    []string `json:"roles" yaml:"roles"`
+}
+
+// Authorizer can auth a subject against a tenant for a permission.
+type Authorizer interface {
+	// Authorize answers the question: can subject S perform permission P on Tenant T?
+	Authorize(subject string, permission Permission, tenant string) bool
+}
```

Comment:
We don't have the consumer yet, so I'm fine having it here. Going foward we should move it to the place it's going to be consumed. :) Do you want to add a `// TODO: ` or good like this?